integration/nevermined/CreditsWithDuration.e2e.test.ts

```diff
@@ -0,0 +1,387 @@
+
+  let salesServices
+  let accessServices
+  let initialBalances: any
+  // let scale: bigint
+
+  // let nft: ethers.Contract
+  let subscriptionNFT: NFT1155Api
+  let neverminedNodeAddress
+
+  let payload: JWTPayload
+
+  before(async () => {
+    TestContractHandler.setConfig(config)
+
+    nevermined = await Nevermined.getInstance(config)
+    ;[, editor, subscriber, , reseller] = await nevermined.accounts.list()
+
+    const clientAssertion = await nevermined.utils.jwt.generateClientAssertion(editor)
+
+    await nevermined.services.marketplace.login(clientAssertion)
+    payload = decodeJwt(config.marketplaceAuthToken)
+
+    assetMetadata = getMetadata()
+    subscriptionMetadata = getMetadata(undefined, 'Subscription Duration NFT1155')
+    subscriptionMetadata.main.type = 'subscription'
+
+    assetMetadata.userId = payload.sub
+    neverminedNodeAddress = await nevermined.services.node.getProviderAddress()
+
+    // conditions
+    ;({ escrowPaymentCondition, transferNftCondition } = nevermined.keeper.conditions)
+
+    // components
+    ;({ token } = nevermined.keeper)
+
+    // scale = 10n ** BigInt(await token.decimals())
+    receivers = [editor.getId(), reseller.getId()]
+
+    // subscriptionPrice1 = subscriptionPrice1 * scale
+    // amounts1 = amounts1.map((v) => v * scale)
+    assetPrice1 = new AssetPrice(
+      new Map([
+        [receivers[0], amounts1[0]],
+        [receivers[1], amounts1[1]],
+      ]),
+    ).setTokenAddress(token.address)
+
+    // subscriptionPrice2 = subscriptionPrice2 * scale
+    // amounts2 = amounts2.map((v) => v * scale)
+
+    assetPrice2 = new AssetPrice(
+      new Map([
+        [receivers[0], amounts2[0]],
+        [receivers[1], amounts2[1]],
+      ]),
+    ).setTokenAddress(token.address)
+
+    royaltyAttributes = getRoyaltyAttributes(nevermined, RoyaltyKind.Standard, royalties)
+
+    initialBalances = {
+      editor: await token.balanceOf(editor.getId()),
+      subscriber: await token.balanceOf(subscriber.getId()),
+      reseller: await token.balanceOf(reseller.getId()),
+      escrowPaymentCondition: Number(await token.balanceOf(escrowPaymentCondition.address)),
+    }
+  })
+
+  describe('As an editor I want to register new content and provide multiple subscriptions to access my content', () => {
+    it('I want to register a subscriptions NFT with multiple plans attached', async () => {
+      console.log(`Running first test`)
+      // Deploy NFT
+      TestContractHandler.setConfig(config)
+
+      const contractABI = await TestContractHandler.getABI(
+        'NFT1155SubscriptionUpgradeable',
+        './test/resources/artifacts/',
+      )
+      subscriptionNFT = await SubscriptionCreditsNFTApi.deployInstance(
+        config,
+        contractABI,
+        editor,
+        [
+          editor.getId(),
+          nevermined.keeper.didRegistry.address,
+          'Credits Subscription NFT',
+          'CRED',
+          '',
+        ],
+      )
```

Comment:
can't we just mint from our 1155 contract?

Reply:
This is a `NFT1155SubscriptionUpgradeable` so no the default `NFT1155Upgradeable` contract. In any case I think for testing purposes is better to deploy the NFT an ensure the environment is clean.